src/dc_tools.rs

```diff
@@ -627,32 +629,94 @@ pub unsafe fn dc_create_outgoing_rfc724_mid(
     let mut rand1: *mut libc::c_char = ptr::null_mut();
     let rand2: *mut libc::c_char = dc_create_id().strdup();
     let ret: *mut libc::c_char;
+    let chatconf: Option<String>;
+    let chatpfx_enabled: bool;
     let mut at_hostname: *const libc::c_char = strchr(from_addr, '@' as i32);
     if at_hostname.is_null() {
         at_hostname = b"@nohost\x00" as *const u8 as *const libc::c_char
     }
+    chatconf = context.get_config(Config::ChatMode);
+    if chatconf.is_none() {
+        chatpfx_enabled = false;
+    } else {
+        if chatconf == Some("1".to_string()) {
+            chatpfx_enabled = true;
+        } else {
+            chatpfx_enabled = false;
+        }
+    }
     if !grpid.is_null() {
-        ret = dc_mprintf(
-            b"Gr.%s.%s%s\x00" as *const u8 as *const libc::c_char,
-            grpid,
-            rand2,
-            at_hostname,
-        )
+        if chatpfx_enabled {
+            ret = dc_mprintf(
```

Comment:
some cleanup would be nice here.
in general, currently, we're cleaning up things before we add new features.

Reply:
Cleanup adressed with commit "Make dc_create_outgoing_rfc724_mid() safe" :)